test/colorado_lottery_test.rb

```diff
@@ -0,0 +1,165 @@
+    lottery = ColoradoLottery.new
+    pick_4 = Game.new('Pick 4', 2)
+    cash_5 = Game.new('Cash 5', 1)
+    mega_millions = Game.new('Mega Millions', 5, true)
+    winston = Contestant.new({
+                         first_name: 'Winston',
+                         last_name: 'Churchill',
+                         age: 18,
+                         state_of_residence: 'CO',
+                         spending_money: 5})
+    frederick = Contestant.new({
+                           first_name:  'Frederick',
+                           last_name: 'Douglas',
+                           age: 55,
+                           state_of_residence: 'NY',
+                           spending_money: 20})
+    alexander = Contestant.new({
+                           first_name: 'Alexander',
+                           last_name: 'Aigades',
+                           age: 28,
+                           state_of_residence: 'CO',
+                           spending_money: 10})
+    lottery.register_contestant(alexander, pick_4)
+    assert_equal ({pick_4 => alexander}), lottery.registered_contestants
+    lottery.register_contestant(alexander, mega_millions)
+    assert_equal ({pick_4 => alexander, mega_millions => alexander }), lottery.registered_contestants
+    lottery.register_contestant(frederick, mega_millions)
+    lottery.register_contestant(winston, cash_5)
+    lottery.register_contestant(winston, mega_millions)
+    assert_equal ({pick_4 => alexander, mega_millions => alexander, mega_millions => frederick, cash_5 => winston, mega_millions => winston}), lottery.registered_contestants
+  end
+
+  def test_it_has_eligible_contestants
+    lottery = ColoradoLottery.new
+    pick_4 = Game.new('Pick 4', 2)
+    cash_5 = Game.new('Cash 5', 1)
+    mega_millions = Game.new('Mega Millions', 5, true)
+    winston = Contestant.new({
+                         first_name: 'Winston',
+                         last_name: 'Churchill',
+                         age: 18,
+                         state_of_residence: 'CO',
+                         spending_money: 5})
+    frederick = Contestant.new({
+                           first_name:  'Frederick',
+                           last_name: 'Douglas',
+                           age: 55,
+                           state_of_residence: 'NY',
+                           spending_money: 20})
+    alexander = Contestant.new({
+                           first_name: 'Alexander',
+                           last_name: 'Aigades',
+                           age: 28,
+                           state_of_residence: 'CO',
+                           spending_money: 10})
+    lottery.register_contestant(alexander, pick_4)
+    lottery.register_contestant(alexander, mega_millions)
+    lottery.register_contestant(frederick, mega_millions)
+    lottery.register_contestant(winston, cash_5)
+    lottery.register_contestant(winston, mega_millions)
+    assert_equal [alexander], lottery.eligible_contestants(pick_4)
```

Comment:
this test is also not thorough enough and doesn't follow the interaction pattern exactly-- the interaction pattern says that pick_4 should return 2 eligible contestants, and also tests some other games to make sure it works correctly.